src/validation.ts

```diff
@@ -0,0 +1,137 @@
+import { Client, cacheExchange, fetchExchange } from '@urql/core';
+import {
+  type DscvrFramesRequest,
+  type DscvrValidationResponse,
+  dscvrClientProtocolPrefix,
+  DscvrUntrustedData,
+} from './types';
+import { DEFAULT_DSCVR_API_URL } from './default';
+
+type ValidationType = string | number | bigint | null | undefined;
+interface ValidatedQueryResult {
+  user: {
+    id: string;
+  };
+  content?: {
+    id: string;
+  };
+  state: string | null;
+  inputText: string | null;
+  timestamp: string;
+  url: string;
+  buttonIndex: number;
+}
+
+const validationQuery = `#graphql
+    query validate($message: String!) {
+      unpackFrameMessage(message: $message) {
+        user {
+            id
+        },
+        content {
+            id
+        }
+        buttonIndex,
+        state,
+        url,
+        timestamp,
+        inputText,
+      }
+    }
+`;
+
+export const validateClientProtocol = (clientProtocol: string) => {
+  return clientProtocol.startsWith(dscvrClientProtocolPrefix);
+};
+
+const validateUntrustedDataValue = (
+  validated: ValidationType,
+  untrusted: ValidationType
+): boolean => {
+  if (!validated && !untrusted) {
+    return true;
+  }
+  if (validated && untrusted && validated === untrusted) {
+    return true;
+  }
+  return false;
+};
+
+const validateUntrustedData = (
+  validatedResult: DscvrValidationResponse,
+  untrustedData: DscvrUntrustedData
+): boolean => {
+  if (
+    validateUntrustedDataValue(
+      validatedResult.validatedDscvrId,
+      untrustedData.dscvrId
+    ) &&
+    validateUntrustedDataValue(
+      validatedResult.validatedContentId,
+      untrustedData.contentId
+    ) &&
+    validateUntrustedDataValue(
+      validatedResult.buttonIndex,
+      untrustedData.buttonIndex
+    ) &&
+    validateUntrustedDataValue(validatedResult.frameUrl, untrustedData.url) &&
+    validateUntrustedDataValue(
+      validatedResult.inputText,
+      untrustedData.inputText
+    ) &&
+    validateUntrustedDataValue(validatedResult.state, untrustedData.state) &&
+    validateUntrustedDataValue(
+      validatedResult.timestamp,
+      Number(untrustedData.timestamp) // TODO: remove this when Proxy is fixed
```

Comment:
not needed, since you already convert to `Number` in `validateFramesPost`

Reply:
No but this is the untrusted, **validateFramesPost**  converts to number the`validatedResult`